bentoml/repository/__init__.py

```diff
@@ -202,15 +204,58 @@ def dangerously_delete(self, bento_name, bento_version):
 
         try:
             response = self.s3_client.delete_object(Bucket=self.bucket, Key=object_name)
-
             DELETE_MARKER = 'DeleteMarker'  # whether object is successfully deleted.
+
+            # Note: as of boto3 v1.13.13. delete_object returns an incorrect format as
+            # expected from documentation.
+            # Expected format:
+            # {
+            #   'DeleteMarker': True|False,
+            #   'VersionId': 'string',
+            #   'RequestCharged': 'requester'
+            # }
+            # Current return:
+            # {
+            #   'ResponseMetadata': {
+            #     'RequestId': '****************',
+            #     'HostId': '*****/******',
+            #     'HTTPStatusCode': 204,
+            #     'HTTPHeaders': {
+            #       'x-amz-id-2': '*****/xxxxx',
+            #       'x-amz-request-id': '332EE9F7AB555D2B',
+            #        'date': 'Tue, 19 May 2020 19:46:57 GMT',
+            #        'server': 'AmazonS3'
+            #     },
+            #     'RetryAttempts': 0
+            #   }
+            # }
+            # An open issue on github: https://github.com/boto/boto3/issues/759
+            if DELETE_MARKER in response:
+                return response[DELETE_MARKER]
```

Comment:
Could we make this and LocalRepository's behavior consistent? The return value ```response ['DELETE_MARKER']``` is not used, let's just use `return` here



```


```

Reply:
Could we add this?
```
if DELETE_MARKER in response and response[DELETE_MARKER] is False:
   logger.warning(f"BentoML has deleted service '{bento_name}:{bento_version}' from YataiService records,
but it failed to delete the saved bundle files stored in s3://{self.bucket}/{object_name}, the
files may have already been deleted by the user.")
```